forage-android/src/test/java/com/joinforage/forage/android/core/element/state/PanElementStateManagerTest.kt

```diff
@@ -13,44 +15,171 @@ class PanSetIsValidTest {
         manager.handleChangeEvent("")
         val state = manager.getState()
         assertThat(state.isValid).isTrue
-        assertThat(state.validationError).isNull()
     }
 
     @Test
-    fun `cardNumber shorter than min 16 length`() {
+    fun `cardNumber too short to contain  IIN`() {
+        val tooShortNoIIN: String = "420"
         val manager = PanElementStateManager.forEmptyInput()
-        manager.handleChangeEvent("12345")
+        manager.handleChangeEvent(tooShortNoIIN)
         val state = manager.getState()
         assertThat(state.isValid).isFalse
-        assertThat(state.validationError).isEqualTo(IncompleteEbtPanError)
     }
 
     @Test
-    fun `cardNumber is min length 16`() {
+    fun `cardNumber has non-existent IIN`() {
+        val invalidIIN: String = "420420420"
         val manager = PanElementStateManager.forEmptyInput()
-        manager.handleChangeEvent("1234567890123456")
+        manager.handleChangeEvent(invalidIIN)
+        val state = manager.getState()
+        assertThat(state.isValid).isFalse
+    }
+
+    @Test
+    fun `cardNumber has valid state IIN but is shorter than expected length`() {
+        val tooShortMaineNumber: String = "507703111" // Maine is 507703
+        val manager = PanElementStateManager.forEmptyInput()
+        manager.handleChangeEvent(tooShortMaineNumber)
+        val state = manager.getState()
+        assertThat(state.isValid).isFalse
+    }
+
+    @Test
+    fun `cardNumber has valid state IIN and is correct length`() {
+        val okMaineNumber: String = "5077031111111111111" // Maine is 507703
+        val manager = PanElementStateManager.forEmptyInput()
+        manager.handleChangeEvent(okMaineNumber)
         val state = manager.getState()
         assertThat(state.isValid).isTrue
-        assertThat(state.validationError).isNull()
+    }
+
+    @Test
+    fun `cardNumber has valid state IIN and is too long`() {
```

Comment:
I blocked the user from being able to type any text longer than the `min(expectedLen, 19)`